src/consent-manager.js

```diff
@@ -127,11 +127,24 @@ export default class ConsentManager {
 	saveConsents() {
 		if (this.consents === null) deleteCookie(this.cookieName);
 		const value = this.config.stringifyCookie(this.consents);
+		const configCookieExpiresAfterDays = this.config.cookieExpiresAfterDays;
+
+		let cookieExpiresAfterDays =
+			typeof configCookieExpiresAfterDays === 'function'
+				? configCookieExpiresAfterDays(this.consents, this.config)
+				: configCookieExpiresAfterDays;
+
+		if (
```

Comment:
I agree with falsies values but `0` as an integer has a meaning for cookie expiration which ends when browser is closed. Using `cookieExpiresAfterDays || 120` will replace `0` to `120` which may not be the expected behavior by the end developer.

To not break existing installations, the default could apply to non-functional `cookieExpiresAfterDays` (like before this feature). When used as a function it'll be up to the developer to explicitly return a value if he don't want a session cookie.

Reply:
I didn't know about this behavior with 0, given the old implementation I thought it was kind of invalid.
I'm kind of torn here, it would be a breaking change to modify the current behavior, but it seems odd to have different behaviors for different methods...